pywhy_graphs/algorithms/tests/test_generic.py

```diff
@@ -39,3 +40,73 @@ def test_convert_to_latent_confounder(graph_func):
     G.remove_edge(3, 1, G.bidirected_edge_name)
     G.add_edge(3, 1, G.directed_edge_name)
     assert pywhy_graphs.is_node_common_cause(G, 3)
+
+
+def test_inducing_path():
+
+    admg = ADMG()
+
+    admg.add_edge("X", "Y", admg.directed_edge_name)
+    admg.add_edge("Z", "Y", admg.bidirected_edge_name)
+    admg.add_edge("Z", "H", admg.bidirected_edge_name)
+
+    # X -> Y <-> z <-> H
+
+    S = {"Y", "Z"}
+    L = set()
+    assert pywhy_graphs.inducing_path(admg, "X", "H", L, S)[0]
+
+    admg.add_edge("H", "J", admg.directed_edge_name)
+
+    # X -> Y <-> z <-> H -> J
+
+    S = {"Y", "Z"}
+    L = {"H"}
+
+    assert pywhy_graphs.inducing_path(admg, "X", "J", L, S)[0]
+
+    admg.add_edge("K", "J", admg.directed_edge_name)
+
+    # X -> Y <-> z <-> H -> J <- K
+    S = {"Y", "Z", "J"}
+    L = {"H"}
+
+    assert not pywhy_graphs.inducing_path(admg, "X", "K", L, S)[0]  # no directed path exists
+
+    admg.add_edge("J", "K", admg.directed_edge_name)
+
+    # X -> Y <-> z <-> H -> J <-> K
+
+    S = {"Y", "J"}
+    L = {"H"}
+
+    assert not pywhy_graphs.inducing_path(admg, "X", "K", L, S)[
+        0
+    ]  # A collider on the path is not in S
+
+    S = {"Y", "Z"}
+    L = set()
+
+    assert not pywhy_graphs.inducing_path(admg, "X", "K", L, S)[
+        0
+    ]  # A non-collider on the path is not in S
+
+
+def test_inducing_path_wihtout_LandS():
```

Comment:
I would also add the test case:

```
A -> B -> C -> D,
B <-> C,
D <-> C
```

and then variants of that where one of the nodes does not satisfy the inducing path definition.